library/analytics/src/main/java/org/cru/godtools/analytics/appsflyer/AppsFlyerAnalyticsService.kt

```diff
@@ -23,23 +30,33 @@ import timber.log.Timber
 
 private const val TAG = "AppsFlyerAnalytics"
 
+@VisibleForTesting
+internal const val AF_DP = "af_dp"
+private const val AF_STATUS = "af_status"
+private const val IS_FIRST_LAUNCH = "is_first_launch"
+
+private const val STATUS_NON_ORGANIC = "Non-organic"
+
 @Singleton
-class AppsFlyerAnalyticsService @Inject internal constructor(private val app: Application, eventBus: EventBus) {
+class AppsFlyerAnalyticsService @Inject internal constructor(
+    private val app: Application,
+    eventBus: EventBus,
+    private val deepLinkResolvers: Set<@JvmSuppressWildcards AppsFlyerDeepLinkResolver>
+) : Application.ActivityLifecycleCallbacks, AppsFlyerConversionListener {
     private val appsFlyer: AppsFlyerLib = AppsFlyerLib.getInstance()
 
     init {
         appsFlyer.apply {
-            init(BuildConfig.APPSFLYER_DEV_KEY, GodToolsAppsFlyerConversionListener, app)
             if (BuildConfig.DEBUG) setLogLevel(AFLogger.LogLevel.DEBUG)
+            init(BuildConfig.APPSFLYER_DEV_KEY, this@AppsFlyerAnalyticsService, app)
+            startTracking(app)
         }
+        app.registerActivityLifecycleCallbacks(this)
+        eventBus.register(this)
 
-        GlobalScope.launch(Dispatchers.Default) {
+        GlobalScope.launch(Dispatchers.IO) {
             val mcId = adobeMarketingCloudId
-            withContext(Dispatchers.Main) {
-                appsFlyer.setAdditionalData(hashMapOf("marketingCloudID" to mcId))
-                appsFlyer.startTracking(app)
-                eventBus.register(this@AppsFlyerAnalyticsService)
```

Comment:
Why don't you need these anymore? Is it something that changed with the upgrade?

Reply:
they got moved a little bit higher in the initialization logic.

basically in order to capture AppsFlyer deep links tracking has to be started, so we can't defer startup until we have the marketing cloud id anymore.

All of this will probably end up changing as we move from Adobe to Google Analytics